### PR TITLE
Stratagem buttons should be disabled based on write locks

### LINKED ISSUE #1152: Stratagem buttons should be disabled based on write locks
The buttons for stratagem should be disabled based on any write lock associated with the stratagem configuration, or the fs.

### PR DESCRIPTION
**Pull request by johnsonw**
Fixes #1152.

The buttons for stratagem should be disabled based on any write lock associated with the stratagem configuration, or the fs.

Signed-off-by: Will Johnson <wjohnson@whamcloud.com>

### CODE REVIEW

**Comment by johnsonw** [comment]
scan-now

**Comment by johnsonw** [comment]
I added the lock on the fsdetail page. The buttons are being disabled as expected but i'm noticing an issue if I scan two filesystems. I get the following error:

```
[2019-08-23 01:44:50,272: ERROR/rpc] RunOneRpc: exception calling run_stratagem: Traceback (most recent call last):

  File "/usr/share/chroma-manager/chroma_core/services/rpc.py", line 107, in run
    "result": self.rpc._local_call(self.body["method"], *self.body["args"], **self.body["kwargs"]),

  File "/usr/share/chroma-manager/chroma_core/services/rpc.py", line 622, in _local_call
    return fn(*args, **kwargs)

  File "/usr/share/chroma-manager/chroma_core/services/job_scheduler/job_scheduler.py", line 1846, in run_stratagem
    client_mount = LustreClientMount.objects.get(host_id=client_host.id)

  File "/usr/lib/python2.7/site-packages/django/db/models/manager.py", line 151, in get
    return self.get_queryset().get(*args, **kwargs)

  File "/usr/lib/python2.7/site-packages/django/db/models/query.py", line 313, in get
    (self.model._meta.object_name, num))

MultipleObjectsReturned: get() returned more than one LustreClientMount -- it returned 2!
```
Currently looking into this.

**Comment by johnsonw** [comment]
disable-by-lock-state

File: iml-wasm-components/iml-stratagem/src/scan_now.rs [hunk 1]
```diff
@@ -138,13 +139,15 @@ fn scan_modal(fs_id: u32, model: &Model) -> Vec<Node<Msg>> {
         class![bs_button::BTN_SUCCESS],
         vec![Node::new_text(if model.disabled {
             "Scanning..."
+        } else if model.is_locked {
+            "Locked 🔒"
```
Comment (jgrund) [inline iml-wasm-components/iml-stratagem/src/scan_now.rs]: Why unicode instead of font-awesome?

**Pull request merged**